From 5766a7164e89b749e92bc2acfe7f46524582efc2 Mon Sep 17 00:00:00 2001
From: Sandor Yu <R01008@freescale.com>
Date: Tue, 7 Apr 2015 14:33:58 +0800
Subject: [PATCH 0342/1221] MLK-10592: dts: Move sii902x property to imx7D dts
 file

sii902x HDMI display is the prime display for imx7d SDB.
so move all sii902x property to imx7d-sdb.dts and
remove imx7d-sdb-hdmi.dts file.

Please make sure imx7D sdb board have added 2.2k pull-up
resistor on the HDMI_I2CSDA and HDMI_I2CSCL signals.
Otherwise kernel will hang at bootup.

Signed-off-by: Sandor Yu <R01008@freescale.com>
[zou:Original patch taken from
git.freescale.com/imx/fsl-arm-yocto-bsp.git-b imx-3.14.52-1.1.0_ga]
Signed-off-by: zou cao <cao.zou@windriver.com>
---
 arch/arm/boot/dts/Makefile           |    1 -
 arch/arm/boot/dts/imx7d-sdb-hdmi.dts |   36 ----------------------------------
 arch/arm/boot/dts/imx7d-sdb.dts      |   19 +++++++++++++++++-
 3 files changed, 18 insertions(+), 38 deletions(-)
 delete mode 100644 arch/arm/boot/dts/imx7d-sdb-hdmi.dts

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index 344b427..c27c910 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -222,7 +222,6 @@ dtb-$(CONFIG_ARCH_MXC) += \
 	imx7d-sdb-enet.dtb \
 	imx7d-sdb-epdc.dtb \
 	imx7d-sdb-gpmi-weim.dtb \
-	imx7d-sdb-hdmi.dtb \
 	imx7d-sdb-qspi.dtb \
 	imx7d-sdb-m4.dtb
 dtb-$(CONFIG_ARCH_MXS) += imx23-evk.dtb \
diff --git a/arch/arm/boot/dts/imx7d-sdb-hdmi.dts b/arch/arm/boot/dts/imx7d-sdb-hdmi.dts
deleted file mode 100644
index 2825e00..0000000
--- a/arch/arm/boot/dts/imx7d-sdb-hdmi.dts
+++ /dev/null
@@ -1,36 +0,0 @@
-/*
- * Copyright (C) 2015 Freescale Semiconductor, Inc.
- *
- * This program is free software; you can redistribute it and/or modify
- * it under the terms of the GNU General Public License version 2 as
- * published by the Free Software Foundation.
- */
-
-#include "imx7d-sdb.dts"
-
-/ {
-	sound-hdmi {
-		compatible = "fsl,imx7d-sdb-sii902x",
-			   "fsl,imx-audio-sii902x";
-		model = "sii902x-audio";
-		cpu-dai = <&sai1>;
-		hdmi-controler = <&sii902x>;
-	};
-};
-
-&clks {
-	assigned-clocks = <&clks IMX7D_PLL_AUDIO_POST_DIV>;
-	assigned-clock-rates = <884736000>;
-};
-
-&sii902x {
-	status = "okay";
-};
-
-&sai1 {
-	assigned-clocks = <&clks IMX7D_SAI1_ROOT_SRC>,
-			  <&clks IMX7D_SAI1_ROOT_CLK>;
-	assigned-clock-parents = <&clks IMX7D_PLL_AUDIO_POST_DIV>;
-	assigned-clock-rates = <0>, <18432000>;
-	status = "okay";
-};
diff --git a/arch/arm/boot/dts/imx7d-sdb.dts b/arch/arm/boot/dts/imx7d-sdb.dts
index 0508492..e059866 100644
--- a/arch/arm/boot/dts/imx7d-sdb.dts
+++ b/arch/arm/boot/dts/imx7d-sdb.dts
@@ -83,6 +83,14 @@
 		codec-master;
 	};
 
+	sound-hdmi {
+		compatible = "fsl,imx7d-sdb-sii902x",
+			   "fsl,imx-audio-sii902x";
+		model = "sii902x-audio";
+		cpu-dai = <&sai1>;
+		hdmi-controler = <&sii902x>;
+	};
+
 	spi4 {
 		compatible = "spi-gpio";
 		pinctrl-names = "default";
@@ -372,7 +380,7 @@
 		mode_str ="1280x720M@60";
 		bits-per-pixel = <16>;
 		reg = <0x39>;
-		status = "disabled";
+		status = "okay";
 	};
 };
 
@@ -765,9 +773,18 @@
 	status = "okay";
 };
 
+&clks {
+	assigned-clocks = <&clks IMX7D_PLL_AUDIO_POST_DIV>;
+	assigned-clock-rates = <884736000>;
+};
+
 &sai1 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_sai1>;
+	assigned-clocks = <&clks IMX7D_SAI1_ROOT_SRC>,
+			  <&clks IMX7D_SAI1_ROOT_CLK>;
+	assigned-clock-parents = <&clks IMX7D_PLL_AUDIO_POST_DIV>;
+	assigned-clock-rates = <0>, <18432000>;
 	status = "okay";
 };
 
-- 
1.7.5.4

